jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -224,6 +230,8 @@ else if (batchBuffer != null && batchSpace >= Generator.MAX_HEADER_LENGTH)
 
         if (flush)
         {
+            for (ByteBuffer bb : buffers)
+                bytesOut.add(bb.limit() - bb.position());
             endPoint.write(this, buffers.toArray(new ByteBuffer[buffers.size()]));
```

Comment:
best practise now is to pass in `new ByteBuffer[0]` to the toArray methods.  It is strangely faster!?!?!